packages/payment-experience-api/src/api/paytrailCardRedirectSuccessController.ts

```diff
@@ -0,0 +1,77 @@
+import { AbstractController, logger } from '@verkkokauppa/core'
+import type { Request, Response } from 'express'
+import { getOrderAdmin } from '@verkkokauppa/order-backend'
+import { URL } from 'url'
+import { getPublicServiceConfiguration } from '@verkkokauppa/configuration-backend'
+import { checkPaytrailCardReturnUrl } from '@verkkokauppa/payment-backend'
+
+export class PaytrailCardRedirectSuccessController extends AbstractController {
+  protected readonly requestSchema = null
+
+  private static success = (url: URL) => {
+    url.pathname = 'success'
+    return url
+  }
+
+  private static fault = (url: URL) => {
+    url.pathname = 'summary'
+    url.searchParams.append('paymentPaid', 'false')
+    return url
+  }
+
+  protected async implementation(req: Request, res: Response): Promise<any> {
+    const globalRedirectUrl = process.env.REDIRECT_PAYTRAIL_PAYMENT_URL_BASE
+    if (!globalRedirectUrl) {
+      throw new Error('No default paytrail redirect url defined')
+    }
+    let redirectUrl = new URL(globalRedirectUrl)
+    try {
+      const orderId = req.params['orderId']
```

Comment:
Could this be destructed from request params?

`const {orderId} = req.params `

Reply:
Why?